### DISCUSSION TITLE
Random trial ordering

### DISCUSSION THREAD

**Discussion by Dawson-Petersen** [immersivecognition/unity-experiment-framework#64] [Q&A] [answered]
Hello! I'm a little new to UXF, so I may just be missing obvious something in the documentation. But I'm trying to figure out how to present my trials in a (semi-) random order within a block. Any recommendations?

**Comment by jackbrookes** [comment, marked as answer]
What is your definition of semi random? If you want them shuffled, you can use the`.Shuffle()` method when building your block. e.g. here:

https://github.com/immersivecognition/unity-experiment-framework/wiki/Factorial-design

```cs
block1.trials.Shuffle();
```

If you want them shuffled, but same for each participant, you can supply an RNG object, the number represents the 'seed':

```cs
var rand = new System.Random(12345); // pick a number to keep consistent
block1.trials.Shuffle(rand);
```

**Reply by Dawson-Petersen** [reply]
Ok, thank you! That helps. I'd like to be able to set up some parameters so that too many trials in the same condition don't appear in a row. Do you know a good way to do that?

**Reply by jackbrookes** [reply]
There's never a clear cut way of doing this - one way would be that you write a function that shuffles - checks if there are too many in a row, then shuffles again, in a loop infinitely. Be careful not to write down impossible conditions or it will run forever!

Or, if you need them same for each participant, just use the 2nd method above and change the number until it meets your condition.

**Reply by Dawson-Petersen** [reply]
Ok, gotcha! Thank you!
